frontend/src/routes/Upload.tsx

```diff
@@ -181,6 +181,7 @@ const UploadMain: React.FC<UploadMainProps> = ({ knownRoles, preselectedSeries }
     const [files, setFiles] = useState<FileList | null>(null);
     const [uploadState, setUploadState] = useRefState<UploadState | null>(null);
     const [metadata, setMetadata] = useRefState<Metadata | null>(null);
+    const selectedFileName = files && files.length > 0 ? files[0].name : null;
```

Comment:
Does `files?.[0]?.name` also work? It's `undefined` instead of `null` in the not-defined case, but should work?